classes/class.ilShortLinkGUI.php

```diff
@@ -0,0 +1,360 @@
+	|                                                                             |
+	| This program is distributed in the hope that it will be useful,             |
+	| but WITHOUT ANY WARRANTY; without even the implied warranty of              |
+	| MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               |
+	| GNU General Public License for more details.                                |
+	|                                                                             |
+	| You should have received a copy of the GNU General Public License           |
+	| along with this program; if not, write to the Free Software                 |
+	| Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA. |
+	+-----------------------------------------------------------------------------+
+*/
+
+require_once('Customizing/global/plugins/Services/UIComponent/UserInterfaceHook/ShortLink/classes/class.ilObjShortLink.php');
+require_once('Customizing/global/plugins/Services/UIComponent/UserInterfaceHook/ShortLink/classes/class.ilShortLinkAccess.php');
+include_once('Customizing/global/plugins/Services/UIComponent/UserInterfaceHook/ShortLink/classes/class.ilShortLinkTableGUI.php');
+require_once('Services/Utilities/classes/class.ilConfirmationGUI.php');
+include_once('Services/Form/classes/class.ilPropertyFormGUI.php');
+require_once('Services/Repository/classes/class.ilObjectPluginGUI.php');
+
+
+/**
+ * GUI-Class ilShortLinkGUI
+ * This class manages the ShortLink frontend.
+ *
+ * @author              Thomas Kolonko <thomas.kolonko@ilub.unibe.ch>
+ * @version             $Id$
+ *
+ * @ilCtrl_isCalledBy   ilShortLinkGUI: ilUIPluginRouterGUI
+ */
+class ilShortLinkGUI extends ilObjectPluginGUI {
+    /**
+     * @var ilTemplate $my_tpl
+     */
+    protected $my_tpl;
+    /**
+     * @var ilCtrl $ctrl
+     */
+    protected $ctrl;
+    /**
+     * @var ilToolbarGUI $toolbar
+     */
+    protected $toolbar;
+    /**
+     * @var ilObjShortLink $obj
+     */
+    protected $obj;
+    /**
+     * @var ilShortLinkAccess $shortLinkAccessChecker
+     */
+    protected $shortLinkAccessChecker;
+    /**
+     * @var ilPropertyFormGUI $form
+     */
+    protected $form;
+    /**
+     * @var ilObjShortLink $externalFeedBlock
+     */
+    protected $externalFeedBlock;
+
+    public function __construct() {
+        global $ilCtrl, $tpl;
+        $this->ctrl = $ilCtrl;
+        $this->form = new ilPropertyFormGUI();
+
+        $this->shortLinkAccessChecker = new ilShortLinkAccess();
+
+        $this->obj = new ilObjShortLink();
+
+        $this->my_tpl = $tpl;
+        $this->my_tpl->getStandardTemplate();
+
+        $this->pl = new ilShortLinkPlugin();
+    }
+
+    /**
+     * @return bool
+     */
+    public function executeCommand() {
+        $cmd = $this->ctrl->getCmd();
+
+        switch($cmd){
+            case 'add':
+            case 'save':
+                $this->$cmd();
```

Comment:
As far as discussed yes, although anonymous certainly makes less to no sense.

Added check for anonymous. Adding new ShortLinks is not possible anymore. Otherwise I suggest we make a ShortLink role in ILIAS (see Bugeno)